src/meta/processors/job/JobDescription.h

```diff
@@ -159,8 +159,7 @@ class JobDescription {
   cpp2::JobDesc toJobDesc();
 
   bool operator==(const JobDescription& that) const {
-    return space_ == that.space_ && type_ == that.type_ && paras_ == that.paras_ &&
-           status_ == that.status_;
```

Comment:
want to double check this is intended

Reply:
Yes.
The jobs in `inFlightJobs_` are all jobs that need to be run and have three states:
`QUEUE`: when adding a job
`FAILED`, `STOPPED`: recover job

So we use `checkOnRunningJobExist` to check if the job in `inFlightJobs_` is running, should not check the status.
For example, there are recover jobs of `FAILED` and `STOPPED` in `inFlightJobs_`. At this time, add the same job again. It was successful in the past, but now we do not allow it because they are the same job.